hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -293,21 +316,16 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(final Lon
 
         setAssignedDistributionSetAndTargetUpdateStatus(assignmentStrategy, distributionSetEntity,
                 targetEntitiesIdsChunks);
+    }
 
-        final Map<String, JpaAction> controllerIdsToActions = createActions(targetsWithActionType, targetEntities,
-                assignmentStrategy, distributionSetEntity);
-        // create initial action status when action is created so we remember
-        // the initial running status because we will change the status
-        // of the action itself and with this action status we have a nicer
-        // action history.
-        createActionsStatus(controllerIdsToActions.values(), assignmentStrategy, actionMessage);
-
-        detachEntitiesAndSendTargetUpdatedEvents(distributionSetEntity, targetEntities, assignmentStrategy);
+    private DistributionSetAssignmentResult buildAssignmentResult(final JpaDistributionSet distributionSet,
+            final Map<String, JpaAction> controllerIdsToActions) {
+        int alreadyAssignedTargetsCount = actionRepository.countByDistributionSetIdAndActiveAndTargetControllerIdNotIn(
```

Comment:
should we return the count of already assigned targets regardless of action status? or should we only return the count of already assigned targets with active actions

Reply:
already assigned means only from the already given collection of targets, no need to make a database call